onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <random>
+
+#include "core/common/logging/logging.h"
+#include "core/framework/utils.h"
+#include "core/graph/graph.h"
+#include "core/providers/xnnpack/xnnpack_execution_provider.h"
+
+#include "test/framework/test_utils.h"
+#include "test/util/include/asserts.h"
+#include "test/util/include/default_providers.h"
+#include "test/util/include/test_utils.h"
+
+#include "gtest/gtest.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace onnxruntime::logging;
+
+#define ORT_MODEL_FOLDER ORT_TSTR("testdata/")
+
+namespace onnxruntime {
+namespace test {
+
+// test uses ONNX model so can't be run in a minimal build.
+// TODO: When we need XNNPACK in a minimal build we should add an ORT format version of the model
+#if !defined(ORT_MINIMAL_BUILD)
+
+// use a snippet from a production model that has NHWC input/output, and Conv nodes with possible Clip and Relu fusion.
+// xnnpack should be able to take all the Conv nodes, and fuse the Conv+Clip and Conv+Relu nodes.
+// That should also mean the Transpose nodes at the start and end of the model can be removed as xnnpack will be
+// handling all other nodes in the model, and the xnnpack nodes will have NHWC input and output.
+TEST(XnnpackEP, TestNhwcConvReluClipFusion) {
+  const ORTCHAR_T* ort_model_path = ORT_MODEL_FOLDER "nhwc_conv_clip_relu.onnx";
+
+  TensorShape input_shape_x{1, 16, 16, 192};
+  auto num_elements = input_shape_x.Size();
+  std::vector<float> input_x;
+  input_x.reserve(num_elements);
+  std::uniform_real_distribution<float> uniform_dist(-128, 128);
+  std::random_device r;
+  std::default_random_engine rd{r()};
+  std::generate_n(std::back_inserter(input_x), input_shape_x.Size(), [&]() -> float { return uniform_dist(rd); });
```

Comment:
consider using [RandomValueGenerator::Uniform()](https://github.com/microsoft/onnxruntime/blob/54d1573d2f6ab27cdf6e1bf5848a51bc03e0853c/onnxruntime/test/common/tensor_op_test_utils.h#L47)
which allows the random seed to be output on failure/set via environment variable